MDBF-883 - CentOS 7 bintar compatibility with old buildbot

RazvanLiviuVarzaru opened the pull request:
As part of the MDBF, I am preparing several (manual) changes to the `centos-7-bintar` container image, one of which will bump `GnuTLS version to 3.7.x,` as in `buildbot.mariadb.net`

`GNUTLS_NO_SIGNAL` flag was introduced in `GnuTLS 3.4.x` so there's no need to disable the macro anymore. (the flag is not present in buildbot.mariadb.net configuration)

More on that, from my tests << 10.11 >>  (thanks @cvicentiu for help)  disabling the macro will cause the compilation to fail when using the newer GnuTLS (static library).
`<command-line>:0:18: error: expected identifier before numeric constant.`

**Important notes:**
This code change should only be deployed in Production when the updated `centos7-bintar` is published in quay, in this order:
  - deploy code changes
  - stop master-docker-nonstandard (where the builder is defined)
  - update image on quay
  - start master-docker-nonstandard

  I will probably deploy this change manually because there are some pending
  other changes which need to be tested and given the BLOCKER
  status of the MDBF, it should be deployed ASAP.

grooverdan reviewed (approved):
I see from MDBF task that the gnutls is a static library build so this looks acceptable.

RazvanLiviuVarzaru commented:
@cvicentiu @grooverdan 
I've added https://github.com/MariaDB/buildbot/pull/684/commits/c9852ceed3296dacb7c14159868710f996b4f453 to solve builds on 11.7/11.8. Please see also the commit message for more details.

grooverdan commented:
if you haven't tested main - after https://github.com/MariaDB/server/pull/3758 might be a good revisit. Some other builders are before gcc 9 so I'm not expecting too much trouble.

RazvanLiviuVarzaru commented:
> @cvicentiu @grooverdan I've added [c9852ce](https://github.com/MariaDB/buildbot/commit/c9852ceed3296dacb7c14159868710f996b4f453) to solve builds on 11.7/11.8. Please see also the commit message for more details.

@cvicentiu 
I removed the commit mentioned above, returned to the old GCC.
You can inspect the following image, look for /changelog

`quay.io/mariadb-foundation/bb-worker:dev_centos7-bintar`

I restricted this builder to run for  >= 11.7
buildbot.mariadb.net does it too:
https://github.com/MariaDB/mariadb.org-tools/commit/437b1f67897766ddde437f14bbbe0b732ef74ddf

RazvanLiviuVarzaru commented:
@cvicentiu I added https://github.com/MariaDB/buildbot/pull/684/commits/9b0e12ded45ba44bab467ae8b57fbe97eb44c837
to remove dead code. Please see the commit message.

cvicentiu reviewed (approved):
Looks good to me.

The schedulers one is a little bit tricky to spot. I realized that the key for those schedulers is for the parent-builder-name to be a substring of the s_packages / s_bigtest builder names.

We really should make that more obvious.
